worker_test.go

```diff
@@ -2,21 +2,380 @@ package main
+					PrivateIp: "1.2.3.4",
+					VpcId:     vpcID,
+					State:     running,
+				},
+				fake.TestInstance{
+					Id:        "i2",
+					Tags:      fake.Tags{"aws:autoscaling:groupName": "asg1", clusterIDTagPrefix + clusterID: "owned"},
+					PrivateIp: "1.2.3.5",
+					VpcId:     vpcID,
+					State:     running,
+				}),
+				DescribeSecurityGroups: fake.R(fake.MockDSGOutput(map[string]string{"id": securityGroupID}), nil),
+				DescribeSubnets: fake.R(fake.MockDSOutput(
+					fake.TestSubnet{Id: "foo1", Name: "bar1", Az: "baz1", Tags: map[string]string{"kubernetes.io/role/elb": ""}}), nil),
+				DescribeRouteTables: fake.R(fake.MockDRTOutput(
+					fake.TestRouteTable{SubnetID: "foo1", GatewayIds: []string{"igw-foo1"}},
+					fake.TestRouteTable{SubnetID: "mismatch", GatewayIds: []string{"igw-foo2"}, Main: true},
+				), nil),
+			},
+			fake.AutoscalingMockOutputs{
+				DescribeAutoScalingGroups: fake.R(fake.MockDASGOutput(map[string]fake.Asgtags{"asg1": {
+					clusterIDTagPrefix + clusterID: "owned",
+				}}), nil),
+				DescribeLoadBalancerTargetGroups: fake.R(&autoscaling.DescribeLoadBalancerTargetGroupsOutput{
+					LoadBalancerTargetGroups: []*autoscaling.LoadBalancerTargetGroupState{},
+				}, nil),
+				AttachLoadBalancerTargetGroups: fake.R(nil, nil),
+			},
+			fake.Elbv2MockOutputs{
+				DescribeTargetGroups: fake.R(nil, nil),
+				DescribeTags:         fake.R(nil, nil),
+			},
+			fake.CfMockOutputs{
+				DescribeStackPages: fake.R(nil, nil),
+				DescribeStacks:     fake.R(nil, nil),
+				CreateStack:        fake.R(fake.MockCSOutput("42"), nil),
+			},
+			awsAdapter.LoadBalancerTypeNetwork,
+		},
+	} {
+		tt.Run(scenario.name, func(t *testing.T) {
+			b, err := os.ReadFile("./testdata/" + scenario.name + "/expected.cf")
+			if err != nil {
+				t.Error(err)
+			}
+			expected := string(b)
+
+			ec2Client := &fake.MockEc2Client{Outputs: scenario.ec2responses}
+			asgClient := &fake.MockAutoScalingClient{Outputs: scenario.asgresponses}
+			elbv2Client := &fake.MockElbv2Client{Outputs: scenario.elbv2responses}
+			cfClient := &fake.MockCloudFormationClient{Outputs: scenario.cfResponses}
+
+			a := &awsAdapter.Adapter{
+				TargetCNI: &awsAdapter.TargetCNIconfig{Enabled: false, TargetGroupCh: make(chan []string, 2)},
+			}
+			a = a.WithCustomAutoScalingClient(asgClient).
+				WithCustomEc2Client(ec2Client).
+				WithCustomElbv2Client(elbv2Client).
+				WithCustomCloudFormationClient(cfClient)
+
+			a, err = a.UpdateManifest(clusterID, vpcID)
+			if err != nil {
+				t.Error(err)
+			}
+
+			f, err := os.Open("./testdata/" + scenario.name + "/ing.yaml")
+			if err != nil {
+				t.Error(err)
+			}
+
+			api, err := kubernetestest.NewAPI(kubernetestest.TestAPIOptions{}, f)
+			if err != nil {
+				t.Error(err)
+			}
+
+			s := httptest.NewServer(api)
+			defer s.Close()
+
+			kubeConfig := kubernetes.InsecureConfig(s.URL)
+			ingressClassFilterList := []string{}
```

Comment:
`ingressClassFilterList := make([]string, 0)`
If you know how many items you will `append`, then you can also use `ingressClassFilterList := make([]string, 0, len(items))` to allocate (`cap`) the underlying array in once.